### DISCUSSION TITLE
New Trigger: Touching

### DISCUSSION THREAD

**Discussion by MarioManX1983** [ikemen-engine/Ikemen-GO#1953] [Feature Request] [answered]
### Is your feature request related to a problem? Please describe.

I've been trying to make interactive stages using the Attached Character feature. And am having severe difficulty with making break able objects. It is possible but it takes a good amount of simi-complex coding to work.

### Describe the solution you'd like

Therefore, I would like to request a new Trigger to take care of this and make it far easier. A trigger that detects if a player, helper, or explod is currently touching another player or helper (Via their blue collision boxes) but not attacking them. I was thinking something like the following.

touching(param_name) = value or "name"

Compatible param_names and values
=========================
player     ;The value can be the player number (1,2,3,4,5,6,7,8), player index, id, or name.
helper     ;The value can be the helper index, id, or name.
explod   ;Value can be the Explod ID.

Example of how it would be used in a State.

[State 0, ChangeAnim]
type = ChangeAnim
trigger1 = touching(player) = 2
value = 2
;elem = 1
ignorehitpause = 1
persistent = 1

This state would change the state owner's animation when the states owner's collision box(s) touches/overlaps player 2s collision box(s).

### Search terms

New Trigger Trigger's Trigger trigger's trigger

**Comment by potsmugen** [comment, marked as answer]
You can do this with the new ClsnVar. Ideally with ZSS loops.

**Reply by MarioManX1983** [reply]
Can you please give me some advice/examples of how to properly use this trigger? I just tried messing around with it and all I could get it to do was to trigger the state whenever the character did a specific action no matter where they were on screen. NOT only when they were touching the object.

Specifically, what I'm trying to do is create an interactive stage with an AttachedChar that has a helper (A barrel in this case) that will change to its breaking animation when any player is touching it while they are in any of their flying back/up or falling down from a hit states. (Like when they are thrown or knocked through the air from a super.)

Also, just as a side note, I don't have experience with zss. Although I do believe I understand the basics after looking at Ikemen Go's files.

**Reply by potsmugen** [reply]
To be honest it's a little complicated if you're not used to ZSS. You'd run one loop through each of player 1's Clsn2, and another through each of player 2's, and see if they overlap.

We _could_ have a "touching" trigger and I even discussed it a few times, but I'm not sure what the syntax could be. It can't be like your suggestion because a char can be touching more than one char so it can't return just one value. I think it'd be something like you provided the ID of the target and Clsn types of player 1 and target to be checked, and it'd return true or false. But still, this can be done with ClsnVar and some elbow grease.
